### PR TITLE
internal/plugins: add `manifests` and `scorecard` plugin keys to all projects

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:**
- internal/plugins: add `manifests` and `scorecard` plugin keys to ansible/v1, helm/v1, and go/v2 projects on init, and consider these keys when running `create api`

**Motivation for the change:** All phase 2 plugins should have keys in the `plugins` config field. This (backwards-compatible) change makes it so.

Signed-off-by: Eric Stroczynski <ericstroczynski@gmail.com>

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: internal/plugins/golang/v2/api.go [hunk 1]
```diff
@@ -46,15 +47,11 @@ func (p *createAPISubcommand) Run() error {
...
+
 	if err := p.CreateAPISubcommand.Run(); err != nil {
 		return err
 	}
 
-	// Emulate plugins phase 2 behavior by checking the config for this plugin's config object.
-	if !hasPluginConfig(p.config) {
-		return nil
-	}
-
```
Comment (camilamacedo86) [inline internal/plugins/golang/v2/api.go:57]: Why we remove it? if is project-version 2 should not add the SDK plugins on top. So, by removing are we not introducing bug?

File: changelog/fragments/phase-2-plugin-keys.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      (ansible/v1, helm/v1, go/v2) Added plugin keys for the `manifests` and `scorecard` phase 2 plugins
+    kind: addition
+    breaking: false
+    migration:
+      header: Add `manifests` and `scorecard` plugin keys to your PROJECT
+      body: >
+        The `manifests` and `scorecard` plugins that create OLM and scorecard manifests, respectively,
+        now have plugin keys that direct `create` subcommands to create related files. While not necessary
+        to add, the following plugin configuration objects will be useful in the future as configuration
+        options are added to their respective plugins:
```
Comment (camilamacedo86) [inline changelog/fragments/phase-2-plugin-keys.yaml:12]:
```suggestion
        options are added to their respective plugins. For Ansible/Helm based-operators update your PROJECT file by adding its plugins:  
```

File: changelog/fragments/phase-2-plugin-keys.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+entries:
+  - description: >
+      (ansible/v1, helm/v1, go/v2) Added plugin keys for the `manifests` and `scorecard` phase 2 plugins
```
Comment (camilamacedo86) [inline changelog/fragments/phase-2-plugin-keys.yaml:3]:
```suggestion
      (ansible/v1, helm/v1, go/v2) Added plugin keys for the `manifests` and `scorecard` phase 2 plugins
```

It is no following the definition described in the template. Also, note that (ansible/v1, helm/v1, go/v2) it is confuse for end-users. We have only one plugin version for Ansible and Helm. So, I think we ought to clarify that by creating 2 entries. Also, note that For Ansible/Helm based-operators it is an additional and for Go-based operators, it is a change.

File: changelog/fragments/phase-2-plugin-keys.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    kind: addition
+    breaking: false
+    migration:
+      header: Add `manifests` and `scorecard` plugin keys to your PROJECT
```
Comment (camilamacedo86) [inline changelog/fragments/phase-2-plugin-keys.yaml:7]:
```suggestion
      header: (For Ansible/Helm based-operators) Update the PROJECT file to add the SDK plugins
```

File: changelog/fragments/phase-2-plugin-keys.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+        ```yaml
+        version: 3-alpha
+        ...
+        plugins:
+          manifests.sdk.operatorframework.io/v2: {}
+          scorecard.sdk.operatorframework.io/v2: {}
+        ```
+
+        For go/v2 projects, you can replace `go.sdk.operatorframework.io/v2-alpha: {}` line with the above two keys.
```
Comment (camilamacedo86) [inline changelog/fragments/phase-2-plugin-keys.yaml:22]:
```suggestion
      Replace that for a new entry which is a change for Go-based operators. 
```

**Review by camilamacedo86 — COMMENTED** [review]
Just a few nits to make the changelog follow up its convention/standard and make it more user friendly and comprehensive to the end-users. Otherwise, It shows great.

File: changelog/fragments/phase-2-plugin-keys.yaml (hunk 3 above)
Comment (estroz) [reply changelog/fragments/phase-2-plugin-keys.yaml:3]: When it comes to changelog entries, it is better to be more specific. I am following the [template](https://github.com/operator-framework/operator-sdk/blob/master/changelog/fragments/00-template.yaml#L10-L11), which says to use your suggestion if the runtime is changing, which it is not.

File: changelog/fragments/phase-2-plugin-keys.yaml (hunk 4 above)
Comment (estroz) [reply changelog/fragments/phase-2-plugin-keys.yaml:7]: This applies to all plugin types.

File: changelog/fragments/phase-2-plugin-keys.yaml (hunk 5 above)
Comment (estroz) [reply changelog/fragments/phase-2-plugin-keys.yaml:22]: This change affects all plugin types. I'd rather not duplicate changelog entries when one will suffice for all projects.

File: changelog/fragments/phase-2-plugin-keys.yaml (hunk 4 above)
Comment (camilamacedo86) [reply changelog/fragments/phase-2-plugin-keys.yaml:7]: Why it applies to all plugins? The golang one has these entries already. See: https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v3/memcached-operator/PROJECT#L12-L14. So, a user with golang based-operator project would need to follow it? So, we are not changing or add anything to go. Am I right?

Comment (camilamacedo86) [reply changelog/fragments/phase-2-plugin-keys.yaml:7]: I catcher: https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v2/memcached-operator/PROJECT#L10-L11 (go/v2) does not do the same as v3.

File: changelog/fragments/phase-2-plugin-keys.yaml (hunk 3 above)
Comment (camilamacedo86) [reply changelog/fragments/phase-2-plugin-keys.yaml:3]: It is true.

File: testdata/go/v2/memcached-operator/PROJECT [hunk 6]
```diff
@@ -8,4 +8,5 @@ resources:
   version: v1alpha1
 version: 3-alpha
 plugins:
-  go.sdk.operatorframework.io/v2-alpha: {}
```
Comment (camilamacedo86) [inline testdata/go/v2/memcached-operator/PROJECT:11]: Here we are changing not adding. So, we might need to a specific entry for go/v2 ( change ) and for helm/ansible an additional.

**Review by camilamacedo86 — APPROVED** [review]
In POV the changelog fragment could be a little more straightforward for end-users and then, shows that for go/v2 it is a change not additional. However, I am fine with this one. Shows a good idea to make all follow the same standard.

**Pull request merged**